diff --git a/Makefile b/Makefile
index a1595a8..1482ead 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-VERSION=v2.4
+VERSION='v2.6, KSV patch 2015-12-15'
 
 prefix=/usr/local
 
diff --git a/librtmp/Makefile b/librtmp/Makefile
index 568fc71..1050efb 100644
--- a/librtmp/Makefile
+++ b/librtmp/Makefile
@@ -1,4 +1,4 @@
-VERSION=v2.4
+VERSION='v2.6, KSV patch 2015-12-15'
 
 prefix=/usr/local
 
