diff --git a/src/plugin.py b/src/plugin.py
index 3418efd..676d535 100644
--- a/src/plugin.py
+++ b/src/plugin.py
@@ -211,10 +211,11 @@ class DevicesMountPanel(Screen, ConfigListScreen):
 			if device in list2:
 				continue
 			for x in mnt:
-				if x.find(device) != -1 and x not in mount_list and '/omb' not in x:
+				if x.find(device) != -1 and '/omb' not in x:
 					parts = x.strip().split()
 					d1 = parts[1]
-					mount_list.append(d1)
+					if d1 not in mount_list and d1 != "/":
+						mount_list.append(d1)
 			self.buildMy_rec(device, mount_list)
 			list2.append(device)
 		f.close()
@@ -448,7 +449,7 @@ class DevicesMountPanel(Screen, ConfigListScreen):
 			extmount = num_mount > 1 and _("%s points") % str(num_mount) or ""
 			res = (fullname, des, png, extmount)
 			self.list.append(res)
-			if num_mount > 1:
+			if num_mount > 0:
 				for mnt in moremount:
 					if str(mnt) != d1:
 						prev_des += '\t' + _("Mount: ") + str(mnt) + '\n' + _("Device: ") + '/dev/' + device + '\t' + _("Type: ") + dtype + rw
@@ -742,9 +743,9 @@ class DevicesMountPanel(Screen, ConfigListScreen):
 						pass
 				f.close()
 				if self.mountp.find('/media/hdd') < 0:
-					if self.mountp != _("None"):
+					if self.mountp != _("None") and self.mountp != "/":
 						system('umount ' + self.mountp)
-					system('umount ' + self.device)
+						system('umount ' + self.device)
 					self.Console.ePopen("/sbin/blkid | grep " + self.device, self.add_fstab, [self.device, self.mountp])
 
 	def add_fstab(self, result=None, retval=None, extra_args=None):
@@ -767,6 +768,8 @@ class DevicesMountPanel(Screen, ConfigListScreen):
 			elif self.device_tmp[3].startswith('UUID='):
 				self.device_uuid = self.device_tmp[3].replace('"', "")
 				self.device_uuid = self.device_uuid.replace('\n', "")
+			else:
+				self.device_uuid = ""
 			if not os.path.exists('/media/hdd'):
 				os.mkdir('/media/hdd', 0o755)
 			flashexpander = None
